sdk/java/dagger-java-annotation-processor/src/main/java/io/dagger/annotation/processor/DaggerModuleAnnotationProcessor.java

```diff
@@ -0,0 +1,433 @@
+    ModuleInfo moduleInfo = generateModuleInfo(annotations, roundEnv);
+
+    Jsonb jsonb = JsonbBuilder.create();
+    String serialized = jsonb.toJson(moduleInfo);
+
+    if (moduleInfo.objects().length == 0) {
+      return true;
+    }
+
+    try {
+      JavaFile f = generate(moduleInfo);
+
+      f.writeTo(processingEnv.getFiler());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    return true;
+  }
+
+  static CodeBlock typeDef(String name) throws ClassNotFoundException {
+    if (name.equals("int")) {
+      return CodeBlock.of(
+          "dag.typeDef().withKind($T.$L)", TypeDefKind.class, TypeDefKind.INTEGER_KIND.name());
+    } else if (name.equals("boolean")) {
+      return CodeBlock.of(
+          "dag.typeDef().withKind($T.$L)", TypeDefKind.class, TypeDefKind.BOOLEAN_KIND.name());
+    } else if (name.startsWith("java.util.List<")) {
+      name = name.substring("java.util.List<".length(), name.length() - 1);
+      return CodeBlock.of("dag.typeDef().withListOf($L)", typeDef(name).toString());
+    } else if (name.endsWith("[]")) {
+      name = name.substring(0, name.length() - 2);
+      return CodeBlock.of("dag.typeDef().withListOf($L)", typeDef(name).toString());
+    }
+
+    var clazz = Class.forName(name);
+    if (clazz.isEnum()) {
+      String typeName = name.substring(name.lastIndexOf('.') + 1);
+      return CodeBlock.of("dag.typeDef().withEnum($S)", typeName);
+    }
+
+    try {
+      if (name.startsWith("java.lang.")) {
+        name = name.substring(name.lastIndexOf('.') + 1);
+      }
+      var kindName = (name + "_kind").toUpperCase();
+      var kind = TypeDefKind.valueOf(kindName);
+      return CodeBlock.of("dag.typeDef().withKind($T.$L)", TypeDefKind.class, kind.name());
+    } catch (IllegalArgumentException e) {
+      String typeName = name.substring(name.lastIndexOf('.') + 1);
+      return CodeBlock.of("dag.typeDef().withObject($S)", typeName);
+    }
+  }
+
+  static Class<?> classForName(String name) throws ClassNotFoundException {
+    if (name.equals("int")) {
+      return int.class;
+    } else if (name.equals("boolean")) {
+      return boolean.class;
+    } else if (name.startsWith("java.util.List<")) {
+      return List.class;
+    } else if (name.endsWith("[]")) {
+      return Class.forName("[L" + name.substring(0, name.length() - 2) + ";");
+    }
+    return Class.forName(name);
+  }
+
+  private String trimDoc(String doc) {
+    if (doc == null) {
+      return null;
+    }
+    return String.join("\n", doc.lines().map(String::trim).toList());
+  }
+
+  private static Boolean isNotBlank(String str) {
+    return str != null && !str.isBlank();
+  }
+
+  private String parseTypeDescription(Element element) {
+    String javadocString = elementUtils.getDocComment(element);
+    if (javadocString == null) {
+      return element.getAnnotation(Module.class).description();
```

Comment:
shouldn't be `element.getAnnotation(Object.class).description();` here? Otherwise it may throw a NPE